typing/types.ml

```diff
@@ -140,6 +140,22 @@ module Variance = struct
   let get_lower v = (mem Pos v, mem Neg v, mem Inv v, mem Inj v)
 end
 
+module Separability = struct
+  type t = Ind | Sep | Deepsep
+  type signature = t list
+  let eq (m1 : t) m2 = (m1 = m2)
+  let rank = function
+    | Ind -> 0
+    | Sep -> 1
+    | Deepsep -> 2
+  let compare m1 m2 = compare (rank m1) (rank m2)
+  let max m1 m2 = if rank m1 >= rank m2 then m1 else m2
```

Comment:
I prefer 

```ocaml
let max m1 m2 = Pervasives.max (rank m1) (rank m2)
```

which makes it clearer that this is just the standard `max`, with nothing funny going on.

Reply:
But this would return the rank, not the mode itself. It would work if we had a `max_by`, but in the meantime I don't see a better way than what we have.

Reply:
Ah, true.  `Pervasives.max` would work, of course, but the code is better as it is.